index.js

```diff
@@ -3,7 +3,7 @@
 var BufferStreams = require('bufferstreams');
 var htmlmin = require('html-minifier');
 var gutil = require('gulp-util');
-var objectAssign = require('object-assign');
```

Comment:
> I'd like to ask you when you use gulp-html without gulp, or even vinyl-fs.

I've never actually used this with gulp. I use gulp for linting and tests, but I use this plugin with assemble, which uses vinyl since we'd like to be supportive of common APIs and plugin signatures. Assemble's community also uses this plugin. I use this plugin, and in the majority of cases extend-shallow will be in my dependency tree when using this. 

> No. As its name suggests

Sounds like you misunderstood, so I'll try to clarify. Seems like you're perspective is based on your gulp toolchain, my isn't. There is no case in which gulp is not a `devDependency` in my projects. My decisions regarding dependencies are entirely predicated on what makes sense for the actual application. I don't waste any time over trying to optimize the `devDependency` tree. 

regarding your point about npm optimizing deps: 

- for object-assign specifically to provide any negligible advantage to support what you're saying, the user would need to be using both npm 3 and gulp 4.
- if different versions of object-assign are in different deps, then both versions will be installed. same result as if it was two completely different libs.